internal/server/v1/event/event.go

```diff
@@ -55,16 +105,42 @@ func Collection(
 		return nil, err
 	}
 
-	resp := &pb.EventCollectionResponse{}
+	resp := &pb.EventCollectionResponse{
+		EventCollection: &pb.EventCollection{
+			Events:         []*pb.EventCollection_Event{},
+			ProvenanceInfo: map[string]*pb.EventCollection_ProvenanceInfo{},
+		},
+	}
 
 	// Go through (ordered) import groups one by one, stop when data is found.
 	for _, btData := range btDataList {
+		if len(btData) == 0 {
+			continue
+		}
+		// Each row represents events from a sub-place. Merge them together.
 		for _, row := range btData {
-			resp.EventCollection = row.Data.(*pb.EventCollection)
-			break
+			data := row.Data.(*pb.EventCollection)
+			for _, event := range data.Events {
+				if filterProp != "" {
+					if !keepEvent(event, filterSpec{
+						prop:  filterProp,
+						unit:  filterUnit,
+						limit: filterLimit,
+					}) {
+						continue
+					}
+				}
+				resp.EventCollection.Events = append(
+					resp.EventCollection.Events,
+					event,
+				)
+			}
+			for provID, info := range data.ProvenanceInfo {
```

Comment:
not sure how likely this would be to happen, but if all events with a specific provID gets filtered out, probably don't want to keep that provID in the provenanceInfo?

Reply:
good point. set provId in event loop now.